tests/test_armv7cpu.py

```diff
@@ -986,6 +986,86 @@ def test_bl_neg(self):
         self.assertEqual(self.rf.read('PC'), pre_pc - 4)
         self.assertEqual(self.rf.read('LR'), pre_pc + 4)
 
+    # CBZ/CBNZ
+
+    @itest_setregs("R0=0")
+    @itest_custom("cbz r0, 0x2a", mode=CS_MODE_THUMB)
+    def test_cbz_taken(self):
+        pre_pc = self.rf.read('PC')
+        self.cpu.execute()
+        self.assertEqual(self.rf.read('PC'), pre_pc + 42)
+
+    @itest_setregs("R0=1")
+    @itest_custom("cbz r0, #0x2a", mode=CS_MODE_THUMB)
+    def test_cbz_not_taken(self):
+        pre_pc = self.rf.read('PC')
+        self.cpu.execute()
+        self.assertEqual(self.rf.read('PC'), pre_pc + 2)
+
+    @itest_setregs("R0=1")
+    @itest_custom("cbnz r0, 0x2a", mode=CS_MODE_THUMB)
+    def test_cbnz_taken(self):
+        pre_pc = self.rf.read('PC')
+        self.cpu.execute()
+        self.assertEqual(self.rf.read('PC'), pre_pc + 42)
+
+    @itest_setregs("R0=0")
+    @itest_custom("cbnz r0, #0x2a", mode=CS_MODE_THUMB)
+    def test_cbnz_not_taken(self):
+        pre_pc = self.rf.read('PC')
+        self.cpu.execute()
+        self.assertEqual(self.rf.read('PC'), pre_pc + 2)
+
+    # TBB/TBH
+
+    @itest_setregs("R0=0xd000", "R1=1")
+    @itest_custom("tbb [r0, r1]", mode=CS_MODE_THUMB)
+    def test_tbb(self):
+        # Write the table of offsets at 0xd000 (R0)
+        # Index is 1 (R1), offset will be 2x21 = 42
+        for i, offset in enumerate([11, 21, 31]):
+            self.mem.write(0xd000 + i, struct.pack('<B', offset))
+
+        pre_pc = self.rf.read('PC')
+        self.cpu.execute()
+        self.assertEqual(self.rf.read('PC'), pre_pc + 4 + 42)
```

Comment:
Why do we do `+4` here?

Reply:
Because `TBB` and `TBH` are 4 bytes long instructions. I have added a comment in my latest commit to hopefully make that clearer.